File: frontend/src/types/session.ts
import type {
  ChatMode,
  ChatResponseStatus,
  SkillUsage,
  StructuredResult,
  UserVisibleError,
} from "./common";

export interface SessionSummary {
  id: string;
  title: string;
  mode: ChatMode | null;
  archived: boolean;
  created_at: string;
  updated_at: string;
}

export interface ChatMessageRecord {
  id: string;
  session_id: string;
  parent_message_id?: string | null;
  role: string;
  content: string;
  mode: ChatMode | null;
  rewritten_query?: string | null;
  skills_used: SkillUsage[];
  result_snapshot?: StructuredResult | null;
  status?: ChatResponseStatus | null;
  user_visible_error?: UserVisibleError | null;
  created_at: string;
}

export interface SessionDetail {
  id: string;
  title: string;
  mode: ChatMode | null;
  archived: boolean;
  created_at: string;
  updated_at: string;
  messages: ChatMessageRecord[];
}

File: backend/app/schemas/session.py
from datetime import datetime
from typing import List, Optional

from pydantic import Field

from .common import (
    ChatMode,
    ChatResponseStatus,
    ContractModel,
    SkillUsage,
    StructuredResult,
    UserVisibleError,
)


class SessionSummary(ContractModel):
    id: str
    title: str
    mode: Optional[ChatMode] = None
    archived: bool = False
    created_at: datetime
    updated_at: datetime


class ChatMessageRecord(ContractModel):
    id: str
    session_id: str
    parent_message_id: Optional[str] = None
    role: str
    content: str
    mode: Optional[ChatMode] = None
    rewritten_query: Optional[str] = None
    skills_used: List[SkillUsage] = Field(default_factory=list)
    result_snapshot: Optional[StructuredResult] = None
    status: Optional[ChatResponseStatus] = None
    user_visible_error: Optional[UserVisibleError] = None
    created_at: datetime


class SessionDetail(ContractModel):
    id: str
    title: str
    mode: Optional[ChatMode] = None
    archived: bool = False
    created_at: datetime
    updated_at: datetime
    messages: List[ChatMessageRecord] = Field(default_factory=list)
